server/src/handler/mod.rs

```diff
@@ -16,6 +19,18 @@ impl<Deps: TpmContextDeps> CommandHandler<Deps> {
     pub fn new(crypto: Deps::Crypto) -> Self {
         Self { crypto }
     }
+    /// invariant: buffer will fit into TPM2B_DIGEST which is much less
+    /// than the maximum request of SP 800 90A
+    fn infallable_get_random(&mut self, buffer: &mut [u8]) {
+        if self.crypto.fill_bytes(&[], buffer).is_err() {
+            let mut entropy = <Deps::Crypto as Drbg>::Entropy::default();
+            self.crypto.fill_entropy(&mut entropy);
+            self.crypto.reseed(&entropy, &[]);
+            let Ok(()) = self.crypto.fill_bytes(&[], buffer) else {
+                unreachable!()
+            };
+        }
+    }
     /// Handles the [TpmCc::GetRandom] (`0x17B`) command.
     pub fn get_random(
```

Comment:
Note that this is going to work until some TPM manufacturer needs to add their own custom command, and then it's going to be a pain (they will have to fork our code, which we should and can avoid IMO).

I'd rather see some more structure or thought around this. Something like trait impls, and then you can register traits as callbacks to command codes. Alternatively if you really want static dispatch (which again I'm not married to that idea, myself), we can do a layered approach to command handling, one that's implemented by us and matches on spec-provided TPM commands, and some injection point elsewhere that is provided by the manufacturer. Anyways, point is this is probably okay for now but it's not a permanent solution.

Plus I don't really want to see all impls for all commands in the same file. We don't need to solve this now, but we do need to think about this, IMO.

Reply:
> Plus I don't really want to see all impls for all commands in the same file. We don't need to solve this now, but we do need to think about this, IMO.

Ofcourse!

Reply:
> Note that this is going to work until some TPM manufacturer needs to add their own custom command, and then it's going to be a pain (they will have to fork our code, which we should and can avoid IMO).

I agree, but I am not sure what would be a good alternative. My intuition tells me there is a way with dyn-dispatch, but I don't know how.

In any case, I will defer work on this for later, also I don't like manual unmarshalling here as well.